From a1a1565766d5f74dac4c7f3fda227dedd01259cd Mon Sep 17 00:00:00 2001
From: Andreas Seltenreich <andreas.seltenreich@credativ.de>
Date: Thu, 3 Mar 2016 19:32:25 +0100
Subject: [PATCH 58/67] CLEANUP: stats: Avoid computation with uninitialized
 bits.

stats_map_lookup() sets bit SMP_F_CONST in the uninitialized member
flags of a stack-allocated sample, leaving the other bits
uninitialized.  All code paths that can access the struct only ever
check for this specific flag, so there is no risk of unintended
behavior.

Nevertheless fix it as it triggers warnings in static code analysis
tools and might become a problem on future revisions of the code.

Problem found in version 1.5.
(cherry picked from commit 9727cf482cc4f761a2649fd04149f17bb3143600)
---
 src/dumpstats.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dumpstats.c b/src/dumpstats.c
index a347a71..4b1b83c 100644
--- a/src/dumpstats.c
+++ b/src/dumpstats.c
@@ -5785,7 +5785,7 @@ static int stats_map_lookup(struct stream_interface *si)
 
 			/* execute pattern matching */
 			sample.data.type = SMP_T_STR;
-			sample.flags |= SMP_F_CONST;
+			sample.flags = SMP_F_CONST;
 			sample.data.u.str.len = appctx->ctx.map.chunk.len;
 			sample.data.u.str.str = appctx->ctx.map.chunk.str;
 			if (appctx->ctx.map.expr->pat_head->match &&
-- 
2.4.10

